docs/design.md

```diff
@@ -0,0 +1,115 @@
+Design notes
+============
+
+Motivation
+----------
+
+To run GitHub Actions self-hosted runners faster and stably by making full use of
+the machine resource in the staging environment.
+
+Goals
+-----
+
+- Deploy and manage GitHub Actions self-hosted runners on multiple bare-metal
+  servers easily with Kubernetes.
```

Comment:
Can this product be used with managed Kubernetes?

Reply:
Not limited to bare-metal k8s, so I'm gonna fix it.